src/Main.elm

```diff
@@ -1,10 +1,571 @@
+
+
+getChildrenNodes : Tree Node -> List Node
+getChildrenNodes tree =
+    tree
+        |> Tree.children
+        |> List.map Tree.label
+
+
+
+-- UPDATE
+
+
+type Msg
+    = NoOp
+    | GoTo Node
+    | GoUp
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        NoOp ->
+            ( model, Cmd.none )
+
+        GoTo node ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    findATree node model.directory
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+        GoUp ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    model.zipperStep
+                        |> Maybe.andThen Zipper.parent
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+
+
+-- VIEW
+
+
+type alias Model =
+    { directory : Tree Node
+    , currentTree : Maybe (Tree Node)
+    , zipperStep : Maybe (Zipper.Zipper Node)
+    }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        nodes : List Node
+        nodes =
+            case model.currentTree of
+                Just tree ->
+                    getChildrenNodes tree
+
+                Nothing ->
+                    []
+    in
+    div [ class "w-screen h-screen bg-gray-50" ]
+        [ div [ class "w-full h-auto" ]
+            [ viewBackButton model
+            , div []
+                (List.append
+                    (viewNodes nodes ViewFolder)
+                    (viewNodes nodes ViewFile)
+                )
+            ]
+        ]
+
+
+viewBackButton : Model -> Html Msg
+viewBackButton model =
+    let
+        hasParent : String
+        hasParent =
+            case parentLabel of
+                Just (Dir id audienceFolder) ->
+                    audienceFolder.name
+
+                Just Root ->
+                    "Home"
+
+                _ ->
+                    ""
+
+        parentLabel : Maybe Node
+        parentLabel =
+            model.zipperStep
+                |> Maybe.andThen Zipper.parent
+                |> Maybe.map Zipper.label
+    in
+    case String.length hasParent > 0 of
+        True ->
+            div [ class "w-full bg-gray-800 px-4 py-4 cursor-pointer", onClick GoUp ]
+                [ span [ class "text-white " ] [ FA.icon FA.arrowAltLeft ]
+                , span [ class "inline-block ml-2 text-white font-bold" ] [ text hasParent ]
+                ]
+
+        False ->
+            text ""
```

Comment:
Random remark: there is a `List.Extra.nothing` helper that does `text ""` under the hood but perhaps tells the story of what's happening better :)

Reply:
Yes absolutely. Anything that documents better :) Btw I couldn't find that function in [List.Extra](https://package.elm-lang.org/packages/elm-community/list-extra/8.5.1/)

Reply:
Oh sorry.. I wrote List.Extra but I meant Html.Extra 🤦